terminus/generators/monolane_generator.py

```diff
@@ -0,0 +1,203 @@
+            # Get a unique and hashable identifier for the street node.
+            node_ident = self.node_ident(street, node)
+            # Ensure it is unique.
+            if node_ident in self.point_name_by_street_node_dict:
+                raise RuntimeError("Non-unique street node identifier. This should not happen.")
+            # Store the point name using the identifier.
+            self.point_name_by_street_node_dict[node_ident] = (point_name, node)
+            # Put the point into the monolane dictionary.
+            new_monolane_point = {
+                point_name: OrderedDict([
+                    ('xypoint', [  # x, y, heading
+                        float(node.center.x),
+                        float(node.center.y),
+                        0  # will be set later using heading between this point and the next
+                    ]),
+                    ('zpoint', [  # z, zdot, theta (superelevation), thetadot
+                        float(node.center.z),
+                        0,
+                        0,
+                        0
+                    ]),
+                ])
+            }
+            # If there was a previous point, calculate the heading from it to this point.
+            if previous_point_name is not None:
+                # Get the previous point's center by name.
+                previous_point = \
+                    self.monolane['maliput_monolane_builder']['points'][previous_point_name]['xypoint']
+                # Calculate the heading.
+                previous_heading = degrees(previous_point_center.yaw(node.center))
+                # Store it.
+                previous_point[2] = previous_heading
+            previous_point_name = point_name
+            previous_point_center = node.center
+            # Insert the new point.
+            self.monolane['maliput_monolane_builder']['points'].update(new_monolane_point)
+        # end for node in list(street.get_nodes())
+        # Update the heading of the last point (last one processed in the for loop).
+        if previous_point_name is not None:
+            # Use the last heading as the heading of the last point.
+            previous_point = \
+                self.monolane['maliput_monolane_builder']['points'][previous_point_name]['xypoint']
+            previous_point[2] = previous_heading
+
+    def start_street(self, street):
+        self.process_street(street)
+
+    def end_street(self, street):
+        pass
+
+    def start_trunk(self, trunk):
+        self.process_street(trunk)
+
+    def end_trunk(self, trunk):
+        pass
+
+    def start_ground_plane(self, plane):
+        pass
+
+    def end_ground_plane(self, plane):
+        pass
+
+    def start_block(self, block):
+        pass
+
+    def end_block(self, block):
+        pass
+
+    def start_building(self, building):
+        pass
+
+    def end_building(self, building):
+        pass
+
+    def to_string(self, stream=None, Dumper=yaml.Dumper, **kwds):
```

Comment:
Even though you don't subclass from `FileGenerator`, could you please implement this in terms of the `generate` / `write_to` public interface? This will make it much easier to refactor later.